### ISSUE TITLE
Certain documented CLI arguments are not recognized in the current release

### ISSUE THREAD

**Issue by coriolinus** [qpdf/qpdf#938]
```bash
$ qpdf --version
qpdf version 11.3.0
Run qpdf --copyright to see copyright and license information.
$ qpdf --annotate=y w8ben.pdf 

qpdf: unrecognized argument --annotate=y

For help:
  qpdf --help=usage       usage information
  qpdf --help=topic       help on a topic
  qpdf --help=--option    help on an option
  qpdf --help             general help and a topic list

$ qpdf --form=y w8ben.pdf

qpdf: unrecognized argument --form=y

For help:
  qpdf --help=usage       usage information
  qpdf --help=topic       help on a topic
  qpdf --help=--option    help on an option
  qpdf --help             general help and a topic list

$ qpdf --modify=form w8ben.pdf
Mon 27 Mar 2023 01:57:37 PM CEST

qpdf: unrecognized argument --modify=form

For help:
  qpdf --help=usage       usage information
  qpdf --help=topic       help on a topic
  qpdf --help=--option    help on an option
  qpdf --help             general help and a topic list

```

However, these arguments are documented to exist:

- `--annotate`: https://qpdf.readthedocs.io/en/stable/cli.html#option-annotate
- `--form`: https://qpdf.readthedocs.io/en/stable/cli.html#option-form
- `--modify`: https://qpdf.readthedocs.io/en/stable/cli.html#option-modify

**Comment by rubyFeedback** [comment]
Perhaps there are too many to show? Perhaps an option "--extended-help" or something could be useful. That is one variant to show all options no matter what.

For larger commandline applications things can become super-unwieldy when doing "--help", so I can understand a focus on more commonly used options; and then perhaps link to the webpage. (Come to think about it, perhaps also an option to display the URLs to these pages, for people who prefer reading online documentation. I usually skip reading local man pages and use the world wide web, unless the --help option is really useful, which was actually the case with qpdf a few months ago when I used it. I try to avoid too "exotic" options myself; I'd never remember them. I always read webpages to update my local knowledge again.)

**Comment by jberkenbilt** [comment]
@rubyFeedback `qpdf --help` is already broken into categories and already references the website.

@coriolinus Those options are only valid inside `--encrypt`. See https://qpdf.readthedocs.io/en/stable/cli.html#encryption

**Comment by jberkenbilt** [comment]
And `qpdf --help=all` shows all options.

**Comment by coriolinus** [comment]
It would be very useful if the documentation at each of these options stated clearly that the option was only valid within `--encrypt`. It's one thing to see that these are "related options" to encryption, but that doesn't teach proper usage when one starts with an encrypted PDF, and wants to modify the document options.

Even better would be if the CLI didn't call these "unrecognized arguments"s, but instead mentioned that these are only valid in conjunction with `--encrypt`.

**Comment by jberkenbilt** [comment]
@coriolinus I agree with both of your suggestions. The first one just requires editing the help text. The second one is quite a bit harder because of the way the command-line parsing works, but it is doable. I've tagged this with enhancement.
